### ISSUE TITLE
Introduce rule that reports missing nullable attributes

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#975]
I'm sure there are other examples, but one case where it would be interesting to report missing nullable attributes is for the lack of **[NotNullWhen(true)]** on the `Equals(object?)` override and the implementation of **IEquatable\<T>**:
```c#
public sealed class Foo : IEquatable<Foo>
{
    public override bool Equals([NotNullWhen(true)] object? obj)
    {
        ...
    }

    public bool Equals([NotNullWhen(true)] Foo? other)
    {
        ...
    }
}
```

The rule should of course also support the explicit interface implementation of **IEquatable\<T>** and only report a diagnostic if NRT is enabled for the method in question.

It should not be reported for the **IEquatable\<T>** implementation if **T** is a value type.

**Comment by drieseng** [comment]
Another - much less common - example would be missing **[MaybeNullWhen(false)]** on **value** argument of an implementation of `IDictionary<TKey,TValue>.TryGetValue(TKey key, [MaybeNullWhen(false)] out TValue value)`.
